apps/spruce/src/pages/task/taskTabs/TaskHistory/CommitDetailsCard/index.tsx

```diff
@@ -0,0 +1,148 @@
+import { useMutation } from "@apollo/client";
+import styled from "@emotion/styled";
+import Badge, { Variant as BadgeVariant } from "@leafygreen-ui/badge";
+import Button, { Size as ButtonSize } from "@leafygreen-ui/button";
+import IconButton from "@leafygreen-ui/icon-button";
+import { palette } from "@leafygreen-ui/palette";
+import { InlineCode } from "@leafygreen-ui/typography";
+import { Link } from "react-router-dom";
+import Icon from "@evg-ui/lib/components/Icon";
+import { size } from "@evg-ui/lib/constants/tokens";
+import { useToastContext } from "@evg-ui/lib/context/toast";
+import { TaskStatus } from "@evg-ui/lib/types/task";
+import { statusColorMap } from "components/TaskBox";
+import { getGithubCommitUrl } from "constants/externalResources";
+import { getTaskRoute } from "constants/routes";
+import {
+  RestartTaskMutation,
+  RestartTaskMutationVariables,
+} from "gql/generated/types";
+import { RESTART_TASK } from "gql/mutations";
+import { useDateFormat, useSpruceConfig } from "hooks";
+import { isProduction } from "utils/environmentVariables";
+import { jiraLinkify, shortenGithash } from "utils/string";
+import { TaskHistoryTask } from "../types";
+
+const { gray } = palette;
+
+interface CommitDetailsCardProps {
+  task: TaskHistoryTask;
+  isCurrentTask: boolean;
+  // TODO DEVPROD-16175: Instead of requesting these fields in TaskHistory query, pass these fields down from the single
+  // Task query. This will reduce duplicated query work since all tasks on the project should come from the same repo.
+  owner: string | undefined;
+  repo: string | undefined;
+}
+
+const CommitDetailsCard: React.FC<CommitDetailsCardProps> = ({
+  isCurrentTask,
+  owner,
+  repo,
+  task,
+}) => {
+  const {
+    canRestart,
+    createTime,
+    displayStatus,
+    id: taskId,
+    order,
+    revision,
+    versionMetadata,
+  } = task;
+  const { author, message } = versionMetadata;
+
+  const spruceConfig = useSpruceConfig();
+  const jiraHost = spruceConfig?.jira?.host ?? "";
+
+  const getDateCopy = useDateFormat();
+  const createDate = new Date(createTime ?? "");
+  const dateCopy = getDateCopy(createDate, {
+    omitSeconds: true,
+    omitTimezone: true,
+  });
+
+  const githubCommitUrl =
+    owner && repo && revision ? getGithubCommitUrl(owner, repo, revision) : "";
+
+  const dispatchToast = useToastContext();
+
+  const [restartTask] = useMutation<
+    RestartTaskMutation,
+    RestartTaskMutationVariables
+  >(RESTART_TASK, {
+    variables: { taskId, failedOnly: false },
+    onCompleted: () => dispatchToast.success("Task scheduled to restart"),
+    onError: (err) =>
+      dispatchToast.error(`Error restarting task: ${err.message}`),
+  });
+
+  return (
+    <CommitCard key={taskId} status={displayStatus as TaskStatus}>
```

Comment:
The commit card should have a blue outline when `isCurrentTask` is true

Reply:
I think it's slightly different; whatever task is currently selected on the timeline will have an outlined corresponding card. We can wire these interactions in [DEVPROD-16185](https://jira.mongodb.org/browse/DEVPROD-16185)